lib/pavilion/test_config/resolver.py

```diff
@@ -1159,3 +1165,21 @@ def resolve_section_values(cls, component, var_man, allow_deferred=False,
             raise TestConfigError("Invalid value type '{}' for '{}' when "
                                   "resolving strings."
                                   .format(type(component), component))
+
+def resolve_cmd_inheritance(test_cfg):
+
+    for section in ['build', 'run']:
+        config = test_cfg.get(section)
+        if not config:
+            continue
+        new_cmd_list = []
+        if config.get('cmds_extend_before'):
```

Comment:
This will still error if the value isn't there, because you'll end up trying to extend a list with `None`.

You want: `config.get('cmds_extend_before', [])`